common/context.go

```diff
@@ -0,0 +1,108 @@
+package common
+
+import (
+    "io"
+    "github.com/inconshreveable/log15"
+    "encoding/json"
+    "bytes"
+    "os"
+)
+
+type PlagiarismConfig struct {
+    Port        string
+    Path        PlagiarismPath
+    FrontendURL string
+    ID          IDConfig
+}
+
+type IDConfig struct {
+    Moss string
+}
+
+type PlagiarismPath struct {
+    Submission string
+    Result string
+}
+
+type TLSConfig struct {
+    CertFile string
+    KeyFile string
+}
+
+type LoggingConfig struct {
+    File string
+    Level string
+}
+
+type Config struct {
+    Plagiarism   PlagiarismConfig
+    TLS          TLSConfig
+    Logging      LoggingConfig
+}
+
+type Context struct {
+    Config      Config
+    Log         log15.Logger
+    handler     log15.Handler
+    logBuffer   *bytes.Buffer
+}
+
+var defaultConfig = Config{
+    Plagiarism: PlagiarismConfig{
+        Path: PlagiarismPath{
+            Submission: "/var/lib/omegaup/submissions",
+            Result: "/var/lib/omegaup/plagiarism",
+        },
+        Port:  "5000",
+        FrontendURL: "https",
+    },
+    TLS: TLSConfig{
+        CertFile: "/etc/omegaup/plagiarism/certificate.pem",
+        KeyFile: "/etc/omegaup/plagiarism/key.pem",
+    },
+    Logging: LoggingConfig{
+        File: "/var/log/omegaup/plagiarism.log",
+        Level: "info",
+    },
+}
+
+
+func NewConfig(reader io.Reader) (*Config, error) {
+    config := defaultConfig
+
+    decoder := json.NewDecoder(reader)
+    if err := decoder.Decode(&config); err != nil {
+        return nil, err
+    }
+    return &config, nil
+}
+
+
+func NewContext(config *Config, role string) (*Context, error) {
```

Comment:
Can you use `github.com/omegaup/quark/common`'s implementation? See https://github.com/omegaup/quark/blob/master/grader/context.go to see how to compose the contexts to avoid code duplication.